subsys/net/lib/dhcpv6/dhcpv6.c

```diff
@@ -1178,6 +1195,42 @@ static int dhcpv6_parse_option_ia_pd(struct net_pkt *pkt, uint16_t length,
 	return 0;
 }
 
+static int dhcpv6_parse_option_dns_servers(struct net_pkt *pkt, uint16_t length,
+					   struct sockaddr_in6 *servers,
+					   uint16_t *server_count)
+
+{
+	const uint8_t addr_size = 16U;
+	uint16_t addr_count;
+	int ret;
+
+	if (length % addr_size != 0) {
+		NET_ERR("Invalid DNS Recursive Name Server option size");
+		return -EMSGSIZE;
+	}
+
+	/* Parse only as many server addresses as the buffer allows. */
+	addr_count = length / addr_size;
+	addr_count = MIN(addr_count, *server_count);
+
+	*server_count = 0;
+	for (int i = 0; i < addr_count; i++) {
+		ret = net_pkt_read(pkt, &servers[i].sin6_addr, addr_size);
+		if (ret < 0) {
+			return ret;
+		}
+
+		length -= addr_size;
```

Comment:
Is it possible that the length could wrap (as it is unsigned)?

Reply:
It shouldn't because the `addr_count` variable, limiting the number of loop iterations, is calculated directly from `length`.